portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -21,29 +21,57 @@
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> messages = new ArrayList<String>();
-  
-  public DataServlet () {
-    messages.add("The sun is out today!");
-    messages.add("I just got new curtains.");
-    messages.add("Woo hoo!");
+  private static final String COMMENT_PARAMETER = "comment-area";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter(COMMENT_PARAMETER);
+    Long timestamp = System.currentTimeMillis();
+
+    Entity bucketListEntity = new Entity("listItem");
+    bucketListEntity.setProperty("item", comment);
+    bucketListEntity.setProperty("time", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
This could be a private member field in this class instead of having to call getDatastoreService in both doPost and doGet.